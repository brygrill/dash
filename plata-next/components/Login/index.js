import React, { PropTypes } from 'react';
import { Grid, Header, Form, Segment } from 'semantic-ui-react';

const Login = (props) => {
  return (
    <div className="plata-height-100">
      <Grid textAlign="center" verticalAlign="middle" className="plata-height-100">
        <Grid.Column className="login-column">
          <Header as="h2" content={props.title} />
          <Form size="large" onSubmit={props.handleSubmit}>
            <Segment>
              <Form.Input
                icon="user"
                iconPosition="left"
                name="email"
                type="email"
                placeholder="Email Address"
                value={props.emailVal}
                onChange={props.emailChange}
              />
              <Form.Input
                icon="lock"
                iconPosition="left"
                name="password"
                type="password"
                placeholder="Password"
                value={props.pwdVal}
                onChange={props.pwdChange}
              />
              <Form.Button fluid primary size="large">Login</Form.Button>
            </Segment>
          </Form>
        </Grid.Column>
      </Grid>
      <style jsx>{`
      .login-column {
        max-width: 450px;
      }
    `}</style>
    </div>
  );
};

Login.defaultProps = {
  title: 'Login',
};

Login.propTypes = {
  title: PropTypes.string,
  emailVal: PropTypes.string.isRequired,
  emailChange: PropTypes.func.isRequired,
  pwdVal: PropTypes.string.isRequired,
  pwdChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Login;
