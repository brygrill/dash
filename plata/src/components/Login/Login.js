import React, { Component } from 'react';
import { Form } from 'semantic-ui-react';
import { login } from '../../firebase/auth';

class Login extends Component {
  constructor() {
    super();
    this.onEmailChange = this.onEmailChange.bind(this);
    this.onPwdChange = this.onPwdChange.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
    this.state = {
      email: '',
      pwd: '',
    };
  }

  onEmailChange(evt, val) {
    this.setState({ email: val.value });
  }

  onPwdChange(evt, val) {
    this.setState({ pwd: val.value });
  }

  onFormSubmit(evt) {
    evt.preventDefault();
    console.log(this.state);
    login(this.state.email, this.state.pwd);
    this.setState({ email: '', pwd: '' });
  }

  render() {
    return (
      <Form onSubmit={this.onFormSubmit}>
        <Form.Group widths="equal">
          <Form.Input
            label="Email:"
            placeholder="email"
            name="email"
            required
            value={this.state.email}
            onChange={this.onEmailChange}
          />
          <Form.Input
            label="Password:"
            placeholder="password"
            name="password"
            type="password"
            required
            value={this.state.pwd}
            onChange={this.onPwdChange}
          />
        </Form.Group>
        <Form.Button>Submit</Form.Button>
      </Form>
    );
  }
}

export default Login;
