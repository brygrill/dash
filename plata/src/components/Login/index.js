import React, { PropTypes } from 'react';
import { Grid, Header, Form, Segment } from 'semantic-ui-react';

const Login = (props) => {
  const styles = {
    maxWidth: props.colWidth,
  };
  return (
    <div className="plata-height-100">
      <Grid textAlign="center" verticalAlign="middle" className="plata-height-100">
        <Grid.Column style={styles}>
          <Header as="h2" content={props.title} />
          <Form size="large" onSubmit={props.handleSubmit}>
            <Segment>
              <Form.Input
                icon="user"
                iconPosition="left"
                name="email"
                type="email"
                placeholder="Email Address"
                value={props.emailVal}
                onChange={props.emailChange}
              />
              <Form.Input
                icon="lock"
                iconPosition="left"
                name="password"
                type="password"
                placeholder="Password"
                value={props.pwdVal}
                onChange={props.pwdChange}
              />
              <Form.Button fluid primary size="large">Login</Form.Button>
            </Segment>
          </Form>
        </Grid.Column>
      </Grid>
    </div>
  );
};

Login.defaultProps = {
  title: 'Login',
  colWidth: '450px',
};

Login.propTypes = {
  title: PropTypes.string,
  colWidth: PropTypes.string,
  emailVal: PropTypes.string.isRequired,
  emailChange: PropTypes.func.isRequired,
  pwdVal: PropTypes.string.isRequired,
  pwdChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Login;
