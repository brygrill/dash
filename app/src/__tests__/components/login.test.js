/* eslint-disable no-console */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Login from '../../components/LoginForm';

const mockFunc = (val) => {
  console.log(val);
};

it('Login component renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Login />, div);
});

test('Login component should render a login form', () => {
  const component = renderer.create(
    <Login
      emailVal={'my@email.com'}
      emailChange={mockFunc}
      pwdVal={''}
      pwdChange={mockFunc}
      handleSubmit={mockFunc}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
