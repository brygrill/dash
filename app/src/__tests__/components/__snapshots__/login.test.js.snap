exports[`test Login component should render a login form 1`] = `
<div
  className="plata-height-100">
  <div
    className="ui center aligned middle aligned grid plata-height-100">
    <div
      className="column"
      style={
        Object {
          "maxWidth": "450px",
        }
      }>
      <h2
        className="ui header">
        Login
      </h2>
      <form
        className="ui large form"
        onSubmit={[Function]}>
        <div
          className="ui segment">
          <div
            className="field">
            <div
              className="ui left icon input">
              <i
                aria-hidden="true"
                className="user icon" />
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                required={undefined}
                type="email"
                value="my@email.com" />
            </div>
          </div>
          <div
            className="field">
            <div
              className="ui left icon input">
              <i
                aria-hidden="true"
                className="lock icon" />
              <input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={undefined}
                type="password"
                value="" />
            </div>
          </div>
          <div
            className="field">
            <button
              className="ui large fluid primary button"
              onClick={[Function]}
              required={undefined}
              tabIndex={undefined}
              type={undefined}>
              Login
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
